app/main/helpers/email.py

```diff
@@ -49,5 +50,12 @@ def decode_email(token):
     ts = URLSafeTimedSerializer(main.config["SECRET_KEY"])
     decoded = ts.loads(token,
                        salt=main.config["RESET_PASSWORD_SALT"],
-                       max_age=86400)
+                       max_age=86400, return_timestamp=True)
     return decoded
+
+
+def token_created_before_password_last_changed(token_timestamp, user):
+
+        password_last_changed = datetime.strptime(
+            user['users']['passwordChangedAt'], '%a, %d %b %Y %H:%M:%S %Z')
```

Comment:
I wonder how brittle this method of decoding the time is?  The API user serialize() method doesn't specify any particular format for the time string - perhaps it should do?  That way we can be sure the timestamp will be in the format expected by the supplier app here.